Ray.jl/src/function_manager.jl

```diff
@@ -0,0 +1,78 @@
+# TODO: this should probably be moved to a Ray.jl package
+
+# NOTES:
+#
+# python function manager maintains a local table of "execution info" with a
+# "function_id" key and values taht are named tuples of name/function/max_calls.
+#
+# python remote function sets a UUID4 at construction time that's used to set
+# the function_hash (???)...comment suggests that "ideally" they'd use the hash
+# of the pickled function but that it's not stable for some reason.
+# but.....neither is a random UUID?????
+#
+# the function table key is built like
+# <key type>:<jobid>:key
+
+# function manager holds:
+# local cache of functions (keyed by function id/hash from descriptor)
+# gcs client
+# maybe job id?
+
+using ray_core_worker_julia_jll: JuliaGcsClient, Exists, Put, Get, JuliaFunctionDescriptor, function_descriptor
+
+# XXX: what's the actual namespace to use?  probably set per-job but I dunno.
+const FUNCTION_MANAGER_NAMESPACE = "JuliaFunctions"
+
+# TODO: remove indirection after fields are settled
+Base.@kwdef struct FunctionManager1
+    gcs_client::JuliaGcsClient
+    functions::Dict{String,Any}
+end
+
+# TODO: remove indirection after fields are settled
+FunctionManager = FunctionManager1
+
+# XXX: we should probably be packaging task-related metadata like the job_id in
+# a task/remotefunction-like struct instead of passing it around like this.
+function_key(fd::JuliaFunctionDescriptor, job_id) = string("RemoteFunction:", job_id, ":", fd.function_hash)
+
+function export_function!(fm::FunctionManager, f, job_id)
+    fd = function_descriptor(f)
+    key = function_key(fd, job_id)
+    if Exists(fm.gcs_client, FUNCTION_MANAGER_NAMESPACE, key, -1)
+        @debug "function already present in GCS store:" key f
+    else
+        @debug "exporting function to GCS store:" key f
+        val = base64encode(serialize, f)
+        Put(fm.gcs_client, FUNCTION_MANAGER_NAMESPACE, key, val, true, -1)
+    end
+end
+
+function wait_for_function(fm::FunctionManager, fd::JuliaFunctionDescriptor, job_id;
```

Comment:
where do we expect to use this?

Reply:
I added it so that I can do a timed wait in the tests; I expect to use it on workers when there may be a race condition vis a vis the functions get exported adn the task being submitted...you don't want to fail out of task execution just because the function hasn't propagated yet.

Reply:
tests are still failing though so my hypothesis that it was a race condition is clearly wrong lol https://github.com/beacon-biosignals/ray_core_worker_julia_jll.jl/actions/runs/5860355101/job/15888248877?pr=19#step:11:167

Reply:
if it's only expected to be used in tests would it be better placed in `runtests.jl`? If we expect to use it in the main source code then ignore.

Reply:
ultimately we _will_ use it in actual code so I think it shoudl stay here (it's based on a similar thing on the python side...)